modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-instance/src/main/java/com/liferay/dynamic/data/mapping/data/provider/instance/DDMStorageTypesDataProvider.java

```diff
@@ -43,11 +48,31 @@ public DDMDataProviderResponse getData(
 
 		List<KeyValuePair> keyValuePairs = new ArrayList<>();
 
+		Optional<HttpServletRequest> httpServletRequestOptional =
+			ddmDataProviderRequest.getParameterOptional(
+				"httpServletRequest", HttpServletRequest.class);
+
+		HttpServletRequest httpServletRequest =
+			httpServletRequestOptional.orElse(null);
+
 		Set<String> storageTypes =
 			ddmStorageAdapterTracker.getDDMStorageAdapterTypes();
 
 		for (String storageType : storageTypes) {
-			keyValuePairs.add(new KeyValuePair(storageType, storageType));
+			if (httpServletRequest == null) {
+				keyValuePairs.add(new KeyValuePair(storageType, storageType));
+
+				continue;
+			}
+
+			String value = LanguageUtil.get(httpServletRequest, storageType);
+
+			if (storageType.equals(StorageType.JSON.toString())) {
+				value = LanguageUtil.get(
+					httpServletRequest, "json[stands-for]", storageType);
+			}
```

Comment:
@carolmariaabb, if the language key were `json` instead of `json[stands-for]` then this `if` statement would be unnecessary, right?

Reply:
@natocesarrego would it be okay to add a new key and maintain `json[stands-for]`? 
I saw that there are other keys, concatenated with `[stands-for]` (`ftl[stands-for]`, `vm[stands-for]`), and they are all being used in the `dynamic-data-mapping-web` module.

Reply:
Thanks for the clarification, @carolmariaabb. Honestly, I don't have this answer right now, because the use of `[stands-for]` on the language keys looks something new to me, therefore we need to confirm if this is intended or not. I'll reach you out in Slack for we discuss about it.

Reply:
Hey @carolmariaabb, after a quick discussion on Slack, I was able to find out the commit which introduced this language key, check it https://github.com/liferay/liferay-portal-ee/commit/b63dc01ef359a605a385e0acfcc2e54dc0919ff0. Therefore, this is a pattern introduced a while ago and we can keep it for now.

I'll move on with the review of this pull.

Thanks.